pkg/controller/volume/persistentvolume/scheduler_binder_test.go

```diff
@@ -1445,38 +1621,54 @@ func TestBindPodVolumes(t *testing.T) {
 		},
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	for name, scenario := range scenarios {
 		klog.V(4).Infof("Running test case %q", name)
 
 		// Setup
 		pod := makePod(nil)
-		if scenario.apiPVs == nil {
-			scenario.apiPVs = scenario.cachedPVs
-		}
+		testEnv := newTestBinder(t, ctx.Done())
 		if scenario.nodes == nil {
 			scenario.nodes = []*v1.Node{node1}
 		}
-		if scenario.provisionedPVCs == nil {
-			scenario.provisionedPVCs = []*v1.PersistentVolumeClaim{}
-		}
-		testEnv := newTestBinder(t)
 		if !scenario.bindingsNil {
 			bindings := []*bindingInfo{}
 			if scenario.binding != nil {
 				bindings = []*bindingInfo{scenario.binding}
 			}
+			claimsToProvision := []*v1.PersistentVolumeClaim{}
+			if scenario.claimToProvision != nil {
+				claimsToProvision = []*v1.PersistentVolumeClaim{scenario.claimToProvision}
+			}
 			testEnv.initNodes(scenario.nodes)
-			testEnv.initVolumes(scenario.cachedPVs, scenario.apiPVs)
-			testEnv.initClaims(scenario.cachedPVCs, nil)
-			testEnv.assumeVolumes(t, name, "node1", pod, bindings, scenario.provisionedPVCs)
+			testEnv.initVolumes(scenario.initPVs, scenario.initPVs)
+			testEnv.initClaims(scenario.initPVCs, scenario.initPVCs)
+			testEnv.assumeVolumes(t, name, "node1", pod, bindings, claimsToProvision)
+		}
+
+		// Before Execute
+		if scenario.apiPV != nil {
```

Comment:
What is the purpose of having a separate call to update the api objects instead of passing them through initVolumes/initClaims?